### PR TITLE
wrap strings with mark_safe for django 2.0 support

### PR DESCRIPTION
**Pull request by vbabiy**

### CODE REVIEW

**Comment by audiolion** [comment]
can you link me to the deprecation warning docs?

**Comment by vbabiy** [comment]
>Support for the allow_tags attribute on ModelAdmin methods is removed.
https://docs.djangoproject.com/en/2.0/releases/2.0/

File: src/auditlog/mixins.py [hunk 1]
```diff
@@ -1,6 +1,12 @@
 import json
 
 from django.conf import settings
+
+try:
+    from django.utils.safestring import mark_safe
+except ImportError:
```
Comment (audiolion) [inline src/auditlog/mixins.py]: Why is this in a try/except? as far as I can tell `django.utils.safestring.mark_safe` is in Django 1.8, 1.9, 1.10, 1.11, and 2.0 with no plans for deprecation.

File: src/auditlog/mixins.py [hunk 2]
```diff
@@ -27,9 +34,10 @@ def user_url(self, obj):
                 link = urlresolvers.reverse(viewname, args=[obj.actor.id])
             except NoReverseMatch:
                 return u'%s' % (obj.actor)
-            return u'<a href="%s">%s</a>' % (link, obj.actor)
+            return mark_safe(u'<a href="%s">%s</a>' % (link, obj.actor))
 
         return 'system'
+
     user_url.allow_tags = True
```
Comment (audiolion) [inline src/auditlog/mixins.py]: remove this line as its not needed since we are adding `mark_safe`

File: src/auditlog/mixins.py [hunk 3]
```diff
@@ -42,7 +50,8 @@ def resource_url(self, obj):
         except NoReverseMatch:
             return obj.object_repr
         else:
-            return u'<a href="%s">%s</a>' % (link, obj.object_repr)
+            return mark_safe(u'<a href="%s">%s</a>' % (link, obj.object_repr))
+
     resource_url.allow_tags = True
```
Comment (audiolion) [inline src/auditlog/mixins.py]: remove this line as its not needed since we are adding `mark_safe`

File: src/auditlog/mixins.py [hunk 4]
```diff
@@ -67,6 +77,7 @@ def msg(self, obj):
             value = [i, field] + (['***', '***'] if field == 'password' else changes[field])
             msg += '<tr><td>%s</td><td>%s</td><td>%s</td><td>%s</td></tr>' % tuple(value)
         msg += '</table>'
-        return msg
+        return mark_safe(msg)
+
     msg.allow_tags = True
```
Comment (audiolion) [inline src/auditlog/mixins.py]: remove this line as its not needed since we are adding `mark_safe`

**Comment by AceFire6** [comment]
Sorry if commenting here is inappropriate, however, I've recently solved this internally due to a recent move to Django 2.0

@vbabiy Instead of marking the returns as safe I believe a better solution would be to use [`format_html`](https://docs.djangoproject.com/en/2.0/ref/utils/#django.utils.html.format_html). The reason for this is that it conditionally escapes all arguments passed into `format_html` unless they are explicitly marked safe and returns a string that is already marked as safe. 

This is what I've done regarding this issue: [gist](https://gist.github.com/AceFire6/ad9d7943f9d0a2f825fc130f923372fe) (excuse the f-strings)

**Comment by theskumar** [comment]
This PR should be closed now? @AceFire6

**Comment by AceFire6** [comment]
@theskumar Yeah, #167 does the same things this does

**Pull request closed without merging**